### ISSUE TITLE
breaks with zstd-1.5.4

### ISSUE THREAD

**Issue by bmwiedemann** [zchunk/zchunk#88]
I'm the packager of `zstd` in openSUSE and we noticed that
https://build.opensuse.org/package/show/Archiving/zchunk breaks with zstd updated from 1.5.2 to 1.5.4

```
[   52s] 25/30 compress auto-chunked file - dict                           FAIL            0.18s   exit status 1
[   52s] >>> LD_LIBRARY_PATH=/home/abuild/rpmbuild/BUILD/zchunk-1.2.3/x86_64-suse-linux/src/lib PATH=/home/abuild/rpmbuild/BUILD/zchunk-1.2.3/x86_64-suse-linux/src/lib:/usr/lib64:/usr/local/bin:/usr/bin:/bin /home/abuild/rpmbuild/BUILD/zchunk-1.2.3/x86_64-suse-linux/test/shacheck src/zck LICENSE.dict.fodt.zck 35549b28b01c0a43d7389d4c40d5b750c73fb6c0c6701355ef709ef4a37df389 -D /home/abuild/rpmbuild/BUILD/zchunk-1.2.3/test/files/LICENSE.dict -o LICENSE.dict.fodt.zck /home/abuild/rpmbuild/BUILD/zchunk-1.2.3/test/files/LICENSE.fodt
[   52s] ――――――――――――――――――――――――――――――――――――― ✀  ―――――――――――――――――――――――――――――――――――――
[   52s] LICENSE.dict.fodt.zck:
[   52s] Calculated checksum: (SHA-256)25669f40abd98cc6a010173ad9891960ef56f12fad2ba95acb642ca11a98e96f
[   52s] Expected checksum: (SHA-256)35549b28b01c0a43d7389d4c40d5b750c73fb6c0c6701355ef709ef4a37df389
[   52s] Checksums don't match!
[   52s] ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
[   52s] 
[   52s] 26/30 decompress auto-chunked file - dict                         OK              0.02s
[   53s] 27/30 compress manual file - no dict                              FAIL            0.23s   exit status 1
[   53s] >>> LD_LIBRARY_PATH=/home/abuild/rpmbuild/BUILD/zchunk-1.2.3/x86_64-suse-linux/src/lib PATH=/home/abuild/rpmbuild/BUILD/zchunk-1.2.3/x86_64-suse-linux/src/lib:/usr/lib64:/usr/local/bin:/usr/bin:/bin /home/abuild/rpmbuild/BUILD/zchunk-1.2.3/x86_64-suse-linux/test/shacheck src/zck LICENSE.manual.nodict.fodt.zck 2580f66412adce1ed8d92abc09b0f64eb155155f61e21a8269a2c4a338857e51 -m -s '<text:' -o LICENSE.manual.nodict.fodt.zck /home/abuild/rpmbuild/BUILD/zchunk-1.2.3/test/files/LICENSE.fodt
[   53s] ――――――――――――――――――――――――――――――――――――― ✀  ―――――――――――――――――――――――――――――――――――――
[   53s] LICENSE.manual.nodict.fodt.zck:
[   53s] Calculated checksum: (SHA-256)f9057c70daf878ec93b86156ccabb4ce4549d409fe0446b4f88a0f88f916956b
[   53s] Expected checksum: (SHA-256)2580f66412adce1ed8d92abc09b0f64eb155155f61e21a8269a2c4a338857e51
[   53s] Checksums don't match!
[   53s] ――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――――
[   53s] 
[   53s] 28/30 decompress manual file - no dict                            OK              0.11s
[   53s] 29/30 compress manual file - dict                                 FAIL            0.21s   exit status 1
[   53s] >>> LD_LIBRARY_PATH=/home/abuild/rpmbuild/BUILD/zchunk-1.2.3/x86_64-suse-linux/src/lib PATH=/home/abuild/rpmbuild/BUILD/zchunk-1.2.3/x86_64-suse-linux/src/lib:/usr/lib64:/usr/local/bin:/usr/bin:/bin /home/abuild/rpmbuild/BUILD/zchunk-1.2.3/x86_64-suse-linux/test/shacheck src/zck LICENSE.manual.dict.fodt.zck 863dc2bda721a32c26485eeb2223d5dabcf9ac517ecba8c7991fa0120d04c937 -D /home/abuild/rpmbuild/BUILD/zchunk-1.2.3/test/files/LICENSE.dict -m -s '<text:' -o LICENSE.manual.dict.fodt.zck /home/abuild/rpmbuild/BUILD/zchunk-1.2.3/test/files/LICENSE.fodt
```

**Comment by jdieter** [comment]
Fixed in 7b84aabb0ff0365e57ae35178eeeb610fc3c0888.  I'll be pushing out a new release shortly.

**Comment by jdieter** [comment]
I've just released 1.2.4 with a fix for this issue.
